parser/metadecoders/decoder.go

```diff
@@ -136,23 +137,30 @@ func (d Decoder) UnmarshalTo(data []byte, f Format, v any) error {
 	case JSON:
 		err = json.Unmarshal(data, v)
 	case XML:
-		var xmlRoot xml.Map
-		xmlRoot, err = xml.NewMapXml(data)
-
-		var xmlValue map[string]any
-		if err == nil {
-			xmlRootName, err := xmlRoot.Root()
-			if err != nil {
-				return toFileError(f, data, fmt.Errorf("failed to unmarshal XML: %w", err))
+		fp := feed.NewParser()
+		if feedData, feedErr := fp.Parse(bytes.NewReader(data)); feedErr == nil {
+			// convert feed parser data to a map
+			d, _ := json.Marshal(feedData)
+			json.Unmarshal(d, v)
```

Comment:
```suggestion
			// We marshal and unmarshal the content, so that the feed data matches the target struct in v.
			var marshalled []byte
			marshalled, err = json.Marshal(feedData)
			if err != nil {
				return toFileError(f, data, fmt.Errorf("temporary JSON marshalling of feed into JSON: %w", err)
			}
			if err := json.Unmarshal(marshalled, v); err != nil {
				return toFileError(f, data, fmt.Errorf("unmarshalling temporary feed JSON: %w", err)
			}
```

I got a bit confused by the marshal/unmarshal pattern first, therefore I also added a comment for future readers.